crates/sui-types/src/messages.rs

```diff
@@ -1594,13 +1595,22 @@ impl Display for MoveLocation {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         let Self {
             module,
-            function,
+            function_idx,
             instruction,
+            message,
         } = self;
-        write!(
-            f,
-            "{module} in function definition {function} at offset {instruction}"
-        )
+        match message {
+            Some(s) => {
+                write!(f, "{s}")
+            }
+
+            None => {
+                write!(
+                    f,
+                    "{module} in function definition {function_idx} at offset {instruction}"
+                )
+            }
+        }
```

Comment:
There's gotta be a more terse way to do this but I'm still a Rust noob 😄

Reply:
This isn't bad :) It's a bit tricky with format strings. With a normal string you could use `Option`'s method `unwrap_or_else`. 

But the format strings aren't really values, so I don't think you can do that. 
You could do `if let Some(s) = message {} else {}` which might at least help reduce horizontal indents if that bothers you